### PR TITLE
create vsam data set

### PR DESCRIPTION
**Pull request by jackjia-ibm**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes https://github.com/zowe/zowe-install-packaging/issues/1688 as research result of https://github.com/zowe/api-layer/issues/889.

#### Changes proposed in this PR

- Place sample JCLs to create VSAM data set into SAMPLIB
- Add new playbook task to create VSAM data set

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function:  Configure Zowe instance *

---
* Description: Provide samples to create VSAM data set for Zowe Caching Service *

---
* Part: SZWESAMP(ZWECSVMP), SZWESAMP(ZWECSVPX)  *

---
When preparing a Zowe instance in High Availability theme, it's required to create
VSAM data set which is used to persist state data for Zowe caching server.
SZWESAMP(ZWECSVSM) is the sample JCL to create the data set. Before 
submitting the job, please read instructions on top of the sample JCL and
customize for your installation.
After this data set is created, the value of instance.env
ZWE_CACHING_SERVICE_VSAM_DATASET variable should be updated to match
the full name of the VSAM data set.

#### Is there a related doc issue or Pull Request? 

Documentation will be added once we have Caching API ready. At this stage, we are not expecting the end user to use this JCL and environment variables.

### CODE REVIEW

**Comment by jackjia-ibm** [comment]
Tests done:
Convenience build: https://wash.zowe.org:8443/view/Zowe/job/zowe-install-test/view/change-requests/job/PR-1760/6/
SMPE: https://wash.zowe.org:8443/view/Zowe/job/zowe-install-test/view/change-requests/job/PR-1760/7/

**Review by stevenhorsman — APPROVED** [review]
Looks plausible to me

File: scripts/instance.template.env [hunk 1]
```diff
@@ -42,6 +42,16 @@ APIML_SECURITY_AUTH_PROVIDER=zosmf
...
 ZWE_DISCOVERY_SERVICES_LIST=https://${ZOWE_EXPLORER_HOST}:${DISCOVERY_PORT}/eureka/
 
+####################
+# caching service
+# TCP port of caching service
+ZWE_CACHING_SERVICE_PORT=7555
+# specify persistent method of caching service
+# currently VSAM is the only option
+ZWE_CACHING_SERVICE_PERSISTENT=VSAM
+# specify the same data set name created by ZWECSVSM jcl
```
Comment (OnnoVdT) [inline scripts/instance.template.env:52]: `# specify the data set name of the caching service VSAM`
you don't specify names of other setup jobs as reference, so don't do it here either

File: files/jcl/ZWECSVSM.jcl [hunk 2]
```diff
@@ -0,0 +1,57 @@
+//ZWECSVSM JOB
```
Comment (OnnoVdT) [inline files/jcl/ZWECSVSM.jcl:1]: Your IDCAMS comments will cause JCL and IDCAMS errors. Also, deleting semi-random lines based on whether you want RLS or not is asking for errors, so it's better to provide two blocks, and a simple switch between them.
I reworked the JCL and attached it here.

[ZWECSVSM.jcl.txt](https://github.com/zowe/zowe-install-packaging/files/5478894/ZWECSVSM.jcl.txt)

**Review by OnnoVdT — CHANGES_REQUESTED** [review]

**Review by OnnoVdT — APPROVED** [review]

**Comment by jackjia-ibm** [comment]
Build has been verified with https://wash.zowe.org:8443/view/Zowe/job/zowe-install-test/view/change-requests/job/PR-1760/9/ and https://wash.zowe.org:8443/view/Zowe/job/zowe-install-test/view/change-requests/job/PR-1760/10/

**Pull request merged**